File: solutions/2089-maximum-matrix-sum/solution.cpp
class Solution {
public:
    long long maxMatrixSum(vector<vector<int>>& matrix) {
        long long sum = 0;
        int mini = INT_MAX;
        int negCount = 0;

        for(int i=0;i<matrix.size();i++){
            for(int j=0;j<matrix[0].size();j++){
                mini =min(mini,abs(matrix[i][j]));
                if(matrix[i][j]<0){
                    negCount++;
                }
                sum+= abs(matrix[i][j]);
            }
        }

        if(negCount%2==0){
            return sum;
        }
        return sum - 2*mini;

    }
};

File: solutions/2089-maximum-matrix-sum/solution.py
class Solution:
    def maxMatrixSum(self, matrix: List[List[int]]) -> int:
        # if odd num are negative, minus the min of abs(matrix[i])
        # if even num are negative, sum of matrix is ans with all pos values
        min_abs = float("inf")
        count_neg = 0
        sumi =0
        for row in matrix:
            for item in row:
                if item < 0:
                    count_neg+=1
                sumi += abs(item)
                min_abs = min(abs(item),min_abs)
        if count_neg%2==0:
            return sumi
        else:
            return sumi-2*min_abs
        
